### DISCUSSION TITLE
Subsite Passwords / Basic Authentication in Quarto

### DISCUSSION THREAD

**Discussion by HaimarR** [quarto-dev/quarto-cli#7511] [Q&A]
### Description

I'm creating a website project in quarto and I want to implement basic authentication. Right now I have a page `index.qmd` with some public content and a button that links to another page `syllabus.qmd`.

For now the content in `syllabus.qmd` is also open to anyone, but I would like to change it so that the page prompts the user to input a password. No username and password stuff, just one password field with a password that is common for all users. For example, if the `syllabus.qmd` password is `1234`, anyone can view the content after inputting `1234`, else it will show a message saying the password is incorrect.

I know how to do this with a local password, but for security reasons I'd like to use server-side authentication. I didn't see any documentation on this in the Quarto documentation website. Any ideas? Thanks!

**Comment by rgaiacs** [comment]
Dear @HaimarR,

authentication is outside of the scope of quarto (command line). Authentication is a feature that should be implemented in [Quarto Pub](https://quarto.org/docs/publishing/quarto-pub.html), [GitHub Pages](https://quarto.org/docs/publishing/github-pages.html), GitLab Pages, [Posit Connect](https://quarto.org/docs/publishing/rstudio-connect.html), [Netlify](https://quarto.org/docs/publishing/netlify.html) or your own web server depending on how you are publishing your document. If you are using your own server, check https://docs.nginx.com/nginx/admin-guide/security-controls/configuring-http-basic-authentication/.

**Comment by mcanouil** [comment]
Related discussions:
- #3816
- #3669
- #6052
